### ISSUE TITLE
Docker Desktop WSL distro stopped on every windows start

### ISSUE THREAD

**Issue by addean** [docker/for-win#8049]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 6EF3A31E-440F-40A1-8C88-3BFE766FDA51/20200813002438

### Actual behavior
Error occurs on every windows start up
### Information

  - Windows Version: 2004
  - Docker Desktop Version: 2.3.0.4
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: None

### Steps to reproduce the behavior
  1. On every windows start up

**Comment by hirsche** [comment]
Same with me after recent windows update

**Comment by hirsche** [comment]
I tried to reinstall everything as well a to install the "Edge"-Version. But nothing worked. I've found some error in the Event-Viewer (it is the German version):

```xml
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
    <System>
        <Provider Name=".NET Runtime" />
        <EventID Qualifiers="0">1026</EventID>
        <Version>0</Version>
        <Level>2</Level>
        <Task>0</Task>
        <Opcode>0</Opcode>
        <Keywords>0x80000000000000</Keywords>
        <TimeCreated SystemTime="2020-08-13T10:04:34.6995804Z" />
        <EventRecordID>39832</EventRecordID>
        <Correlation />
        <Execution ProcessID="0" ThreadID="0" />
        <Channel>Application</Channel>
        <Computer>DESKTOP-FHNFD6Q</Computer>
        <Security />
    </System>
    <EventData>
        <Data>
            Anwendung: Docker Desktop.exe Frameworkversion: v4.0.30319 Beschreibung: Der Prozess wurde aufgrund einer unbehandelten Ausnahme beendet. Ausnahmeinformationen: System.UnauthorizedAccessException bei System.IO.__Error.WinIOError(Int32, System.String) bei System.IO.FileStream.Init(System.String, System.IO.FileMode, System.IO.FileAccess, Int32, Boolean, System.IO.FileShare, Int32, System.IO.FileOptions, SECURITY_ATTRIBUTES, System.String, Boolean, Boolean, Boolean) bei System.IO.FileStream..ctor(System.String, System.IO.FileMode, System.IO.FileAccess, System.IO.FileShare, Int32, System.IO.FileOptions, System.String, Boolean, Boolean, Boolean) bei System.IO.StreamWriter.CreateFile(System.String, Boolean, Boolean) bei System.IO.StreamWriter..ctor(System.String, Boolean, System.Text.Encoding, Int32, Boolean) bei System.IO.StreamWriter..ctor(System.String, Boolean) bei System.IO.File.CreateText(System.String) bei Docker.Core.Cli.CliOptionsLoader.Save(Docker.Core.CliOptions) bei Docker.ApiServices.ClientConfig.DockerLoginSync.CredentialChanged(System.String, Docker.Core.Credential) bei Docker.ApiServices.Registries.DockerHub+
            <FinalizeLogin>
                d__20.MoveNext() bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task) bei Docker.ApiServices.Registries.DockerHub+
                <LogIn>
                    d__17.MoveNext() bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task) bei Docker.ApiServices.Registries.DockerHub+
                    <AutoLogIn>d__16.MoveNext() bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() bei System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean) bei System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean) bei System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem() bei System.Threading.ThreadPoolWorkQueue.Dispatch()
        </Data>
    </EventData>
</Event>
```
and

```xml
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
    <System>
        <Provider Name="Application Error" />
        <EventID Qualifiers="0">1000</EventID>
        <Version>0</Version>
        <Level>2</Level>
        <Task>100</Task>
        <Opcode>0</Opcode>
        <Keywords>0x80000000000000</Keywords>
        <TimeCreated SystemTime="2020-08-13T10:04:34.9335808Z" />
        <EventRecordID>39833</EventRecordID>
        <Correlation />
        <Execution ProcessID="0" ThreadID="0" />
        <Channel>Application</Channel>
        <Computer>DESKTOP-FHNFD6Q</Computer>
        <Security />
    </System>
    <EventData>
        <Data>Docker Desktop.exe</Data>
        <Data>2.3.0.46911</Data>
        <Data>d315d98b</Data>
        <Data>KERNELBASE.dll</Data>
        <Data>10.0.19041.423</Data>
        <Data>ed02c0bc</Data>
        <Data>e0434352</Data>
        <Data>0000000000023e49</Data>
        <Data>2128</Data>
        <Data>01d67159218ddcc9</Data>
        <Data>C:\Program Files\Docker\Docker\Docker Desktop.exe</Data>
        <Data>C:\WINDOWS\System32\KERNELBASE.dll</Data>
        <Data>78c3ad35-2c4f-4b79-9f9c-eea0dd6d3817</Data>
        <Data />
        <Data />
    </EventData>
</Event>
```

Please, I really need urgent help, because otherwise I cannot work with docker anymore.

**Comment by hirsche** [comment]
Ok at least for me the problem got solved. I don't know why this happened, but one day the docker daemon did not start but gave me the hint that my `$env:USERPROFILE\.docker\config.json` is not writable. Thus, the daemon tried to alter the file, however, it was read only.
Unfortunately, this was also something I caused, because I've changed it to read-only. Some application always tried to add google credentials which cause docker-compose and docker commands to fail and I tried to find out which one. But, I've forgot about that and docker did not really explain in the error message (of the event-log) why it was failing and what was actually not accessible.
Finally making `config.json` readable again fixed the `UnauthorizedAccessException`.
